snoop/data/analyzers/email.py

```diff
@@ -26,14 +27,56 @@ def get_headers(message):
     return dict(rv)
 
 
+def dump_part(message):
+    rv = {'headers': get_headers(message)}
+
+    content_type = message.get_content_type()
+    if content_type == 'text/plain':
+        payload_bytes = message.get_payload(decode=True)
+        charset = message.get_content_charset() or 'latin1'
+        rv['text'] = payload_bytes.decode(charset, errors='replace')
+
+    if content_type == 'text/html':
+        payload_bytes = message.get_payload(decode=True)
+        with models.Blob.create() as writer:
+            writer.write(payload_bytes)
+        rmeta_blob = tika.rmeta(writer.blob)
+        with rmeta_blob.open(encoding='utf8') as f:
+            rmeta_data = json.load(f)
+        rv['text'] = rmeta_data[0]['X-TIKA:content']
+
+    if message.is_multipart():
+        rv['parts'] = [dump_part(part) for part in message.get_payload()]
```

Comment:
If you recurse into `dump_part` for parts of the message, `dump_part` will try to extract the headers from those: see the first line from `dump_part`. I don't think it's a good idea to look for headers where there shouldn't be any.

Reply:
We're going to need headers of sub-parts when we're going to decrypt pgp emails.